### PR TITLE
43 connect database (Domain)

### LINKED ISSUE #43: connect database (Domain)

### PR DESCRIPTION
**Pull request by dottore1**
add batchRepository interface
add machineRepository interface
annotate machine to jpa enity
annotate batch to jpa entity
remove design mistakes in database table design
annotate brewmes as service
implement use of  MachineRepository in BrewMes
implement MachineState enum
make database use UUID 

Closed #43

### CODE REVIEW

**Review by Lawst96 — CHANGES_REQUESTED** [review]
For the uninitiated, a lot of these annotations and methods don't make a lot of sense. Consider adding comments to make their purpose more clear.

File: src/main/java/com/BrewMES/demo/model/BrewMES.java [hunk 1]
```diff
@@ -1,37 +1,32 @@
...
-	}
-	//Singleton method to get the BrewMes instance.
-	public static BrewMES getInstance() {
-		if (instance == null) {
-			instance = new BrewMES();
-		}
-		return instance;
+	public BrewMES() {
+
 	}
```
[the remark this answers is not in the document]
Comment (dottore1) [reply src/main/java/com/BrewMES/demo/model/BrewMES.java]: Sure. We don't need to state that explicit at least. Removed in -> https://github.com/Semester3-03/Java-application/pull/87/commits/81b3a2b747fe0983f1997eae68d22e8f4966798c

File: src/main/java/com/BrewMES/demo/model/BrewMES.java [hunk 2]
```diff
@@ -96,8 +94,17 @@ public String getMachineVariables() {
 		throw new UnsupportedOperationException();
 	}
 
+	//checks if not each machine present in the database also is present in the machines map,it puts the machine there.
```
[the remark this answers is not in the document]
Comment (dottore1) [reply src/main/java/com/BrewMES/demo/model/BrewMES.java]: Rewrote that in ->https://github.com/Semester3-03/Java-application/pull/87/commits/81b3a2b747fe0983f1997eae68d22e8f4966798c

File: src/main/java/com/BrewMES/demo/model/BrewMES.java [hunk 3]
```diff
@@ -96,8 +99,20 @@ public String getMachineVariables() {
 		throw new UnsupportedOperationException();
 	}
 
+	/**
+	 * Finds all machines in the database and updates the the Map machines.
```
Comment (Lawst96) [inline src/main/java/com/BrewMES/demo/model/BrewMES.java:103]: "the the"
```suggestion
	 * Finds all machines in the database and updates the Map machines.
```

**Review by Lawst96 — CHANGES_REQUESTED** [review]

**Review by Lawst96 — APPROVED** [review]

**Review by Kaszz — APPROVED** [review]
🐸

File: src/main/java/com/BrewMES/demo/API/BrewMESController.java [hunk 4]
```diff
@@ -1,11 +1,9 @@
 package com.BrewMES.demo.API;
 
-import com.BrewMES.demo.model.BeerType;
-import com.BrewMES.demo.model.BrewMES;
-import com.BrewMES.demo.model.Command;
-import com.BrewMES.demo.model.iBrewMES;
+import com.BrewMES.demo.model.*;
```
Comment (Teyson) [inline src/main/java/com/BrewMES/demo/API/BrewMESController.java:3]: Jeg kan ikke helt se hvorfor BrewMESController har brug for alt i model pakken? Skal den ikke kun bruge IBrewMES?

**Review by Teyson — CHANGES_REQUESTED** [review]
Udover min ene meget lille kommentar ser det rigtig fint ud :)

File: src/main/java/com/BrewMES/demo/API/BrewMESController.java (hunk 4 above)
Comment (dottore1) [reply src/main/java/com/BrewMES/demo/API/BrewMESController.java:3]: Det har du ret i, men jeg tror det hører til et andet issue, fordi vi stadig bruger enums i controlleren. Det blev vi enige om, at fjerne, så BrewMes har ansvaret for det

**Review by Wylardt — APPROVED** [review]

**Review by Lise4070 — APPROVED** [review]

**Review by Teyson — APPROVED** [review]

**Pull request merged**